src/chain/block.cpp

```diff
@@ -859,16 +857,16 @@ code block::accept(const chain_state& state,
     const system::settings& settings, bool transactions, bool header) const
 {
     code ec;
-    const auto bip16 = state.is_enabled(rule_fork::bip16_rule);
-    const auto bip34 = state.is_enabled(rule_fork::bip34_rule);
-    const auto bip113 = state.is_enabled(rule_fork::bip113_rule);
-    const auto bip141 = state.is_enabled(rule_fork::bip141_rule);
+    const auto bip16 = state.is_enabled(bip16_rule);
+    const auto bip34 = state.is_enabled(bip34_rule);
+    const auto bip113 = state.is_enabled(bip113_rule);
+    const auto bip141 = state.is_enabled(bip141_rule);
 
     const auto max_sigops = bip141 ? max_fast_sigops : max_block_sigops;
     const auto block_time = bip113 ? state.median_time_past() :
         header_.timestamp();
 
-    if (header && (ec = header_.accept(state)))
+    if (header && ((ec = header_.accept(state))))
```

Comment:
Good catch. I generally try to avoid this idiom except in the simple case of a single condition.